Assets/00_Content/Scripts/Interactables/Table.cs

```diff
@@ -39,7 +43,71 @@ public class Table : Interactable {
 		}
 
 		public override bool CanInteract(GameObject player, PickUp item) {
-			return !IsDestroyed && player.GetComponent<PlayerSteward>().Follower != null;
+			if (!IsDestroyed && player.GetComponent<PlayerSteward>().Follower != null)
+				return true;
+
+			if (IsDestroyed && item is RepairTool) {
+				if (Tavern.Instance != null && RoundController.Instance != null &&
+					Tavern.Instance.Money < RoundController.Instance.CurrentDifficulty.tableRepairCost) {
+					return false;
+				}
+
+				return true;
+			}
+
+			return false;
+		}
+
+		private void Update() {
+			UpdateRepairing();
+		}
+
+		public override void Interact(GameObject player, PickUp item) {
+			if (IsDestroyed && item is RepairTool tool) {
+				StartRepairing(player, tool);
+			}
+		}
+
+		public override void CancelInteraction(GameObject player, PickUp item) {
+			EndRepairing(player);
+		}
+
+		private void StartRepairing(GameObject player, RepairTool tool) {
+			isRepairing = true;
+			repairTool = tool;
+			repairTimer = repairDuration = RoundController.Instance != null
+				? RoundController.Instance.CurrentDifficulty.tableRepairTime
+				: 5f;
+			tool.RepairProgressImage.fillAmount = 0f;
+			tool.RepairProgressCanvas.SetActive(true);
+			tool.RepairProgressCanvas.transform.SetPositionAndRotation(
+				transform.position + new Vector3(0, 3f, 0), Quaternion.identity);
```

Comment:
Would it be more fitting to have this logic in the repairtool? It would make creating another repairable easier.